src/app/companies-add/companies-add.component.ts

```diff
@@ -0,0 +1,121 @@
+import { Component, ElementRef, OnInit, ViewChild } from '@angular/core';
+import { HttpClient } from "@angular/common/http";
+import { CompaniesAddService } from "./companies-add.service";
+import { Observable, ReplaySubject, Subject } from "rxjs";
+import { FormControl, FormGroup, Validators } from "@angular/forms";
+import { map, startWith } from 'rxjs/operators';
+import { MatDialog } from '@angular/material/dialog';
+import { DialogComponent } from "../dialog/dialog.component";
+import { ActivatedRoute } from "@angular/router";
+import {AuthService} from "../auth/auth.service";
+
+interface UNIVERSITY {
+  entity: string,
+  //university: string
+}
+
+@Component({
+  selector: 'app-company-reg',
+  templateUrl: './companies-add.component.html',
+  styleUrls: ['./companies-add.component.css']
+})
+export class CompaniesAddComponent implements OnInit {
+
+  entities: string[] = [];
+  // universities: UNIVERSITY[] = [];
+  // countries = {
+  //   raw: [] as string[],
+  //   filtered: null as null | Observable<string[]>
+  // };
+  // years = ["School Leaver", "1st Year", "2nd Year", "3rd Year", "4th Year", "Graduate"];
+
+  form = new FormGroup({
+    company_name: new FormControl(null, [Validators.required]),
+    company_address: new FormControl(null, [Validators.required]),
+    company_email: new FormControl(null, [Validators.required, Validators.email]),
+    company_phone: new FormControl(null, [Validators.required,
+    Validators.pattern('(^(0\\s*)([0-9]\\s*){8}[0-9]$)|^\\+([0-9]\\s*)*$')]),
+    full_name: new FormControl(null, [Validators.required]),
+    designation: new FormControl(null, [Validators.required]),
+    email: new FormControl(null, [Validators.required, Validators.email]),
+    phone: new FormControl(null, [Validators.required,
+    Validators.pattern('(^(0\\s*)([0-9]\\s*){8}[0-9]$)|^\\+([0-9]\\s*)*$')]),
+    entity: new FormControl()
+  });
+
+  public filteredUniversities: UNIVERSITY[] = [];
+
+  // @ts-ignore
+  @ViewChild('universitySearch') inputEl: ElementRef;
+
+  constructor(private signUpService: CompaniesAddService, private dialog: MatDialog, private route: ActivatedRoute) {
+  }
+
+  async ngOnInit() {
+
+    if (<string>this.route.snapshot.paramMap.get("entity")) {
+      this.form.get("entity")?.setValue(<string>this.route.snapshot.paramMap.get("entity"));
+    }
+
+    // this.form.get("entity")?.setValue("admin");
+    // this.form.get("entity")?.setValue(<string>this.authService.getEntity());
+    // console.log("Role: ", this.authService.getRole());
+    // console.log("Entity: ", this.authService.getEntity());
+
+    this.signUpService.getEntities().subscribe(data => {
+      this.entities = data;
+    });
+
+  }
+
+  isFormValid(): boolean {
+    if (!this.form.valid) return false;
+    return true;
+  }
+
+  randomNumberInRange() {
```

Comment:
We will have to change the logic here. Since we don't have a ton of entities, randomizer can be biased in short term and in small quantities. Entities might not be very keen on this. We should have a counter and use it.

Reply:
No changes made to this till finalized as we discussed